contracts/System.sol

```diff
@@ -80,4 +95,118 @@ contract System {
+    return _govHubAddr();
+  }
+  // -------
+
+  function _send(address sendTo, uint256 amount) internal returns (bool) {
+    if (!_sufficientBalance(amount)) {
+      return false;
+    }
+    return payable(sendTo).send(amount);
+  }
+
+  function _unsafeSend(address sendTo, uint256 amount) internal returns (bool) {
+    // internally invokes call() with no gas limit! should always be reentrancy safe
+    if (!_sufficientBalance(amount)) {
+      return false;
+    }
+    (bool success, ) = payable(sendTo).call{value: amount}("");
+    return success;
+  }
+
+  function _transfer(address sendTo, uint256 amount) internal {
+    require(_sufficientBalance(amount), "insufficient balance");
+    payable(sendTo).transfer(amount);
+  }
+
+  function _unsafeTransfer(address sendTo, uint256 amount) internal {
+    // internally invokes call() with no gas limit! should always be reentrancy safe
+    Address.sendValue(payable(sendTo), amount);
+  }
+
+  function _sufficientBalance(uint256 amount) private view returns (bool) {
+    return address(this).balance >= amount;
+  }
+
+
+
+
+  // --- Registry functions below are platform-contracts that are safe to access ---
+
+  function _foundationPayable() internal returns(address payable) {
+    _cacheAllContractsLocally();
+    return payable(s_allContracts.foundationAddr);
+  }
+
+  function _systemReward() internal returns(ISystemReward) {
+    _cacheAllContractsLocally();
+    return s_allContracts.systemReward;
+  }
+
+  function _systemRewardPayable() internal returns(address payable) {
+    _cacheAllContractsLocally();
+    address _systemRewardAddr = address(s_allContracts.systemReward);
+    return payable(_systemRewardAddr);
+  }
+
+  function _lightClient() internal returns(ILightClient) {
+    _cacheAllContractsLocally();
+    return s_allContracts.lightClient;
+  }  
+
+  function _relayerHub() internal returns(IRelayerHub) {
+    _cacheAllContractsLocally();
+    return s_allContracts.relayerHub;
+  }
+
+  function _candidateHub() internal returns(ICandidateHub) {
+    _cacheAllContractsLocally();
+    return s_allContracts.candidateHub;
+  }  
+
+  function _pledgeAgent() internal returns(IPledgeAgent) {
+    _cacheAllContractsLocally();
+    return s_allContracts.pledgeAgent;
+  }
+
+  function _burnContract() internal returns(IBurn) {
+    _cacheAllContractsLocally();
+    return s_allContracts.burn;
+  }
+
+  function _validatorSet() internal returns(IValidatorSet) {
+    _cacheAllContractsLocally();
+    return s_allContracts.validatorSet;
+  }  
+
+  function _slashIndicator() internal returns(ISlashIndicator) {
+    _cacheAllContractsLocally();
+    return s_allContracts.slashIndicator;
+  }  
+
+  function _govHubAddr() internal returns(address) {
+    _cacheAllContractsLocally();
+    return s_allContracts.govHubAddr;
+  }
+
+  function _cacheAllContractsLocally() private {
+    //@correlate-registry.cache
+    if (s_allContracts.burn == IBurn(address(0))) { 
```

Comment:
Should be fine. Had a thought though: what if a future CoreSDK blockchain forks this codebase and removes the `Burn` contract because they don't want the burn functionality? I guess it's reasonable for it to be on them to resolve this, as any other approach results in us expensively checking a lot more storage.